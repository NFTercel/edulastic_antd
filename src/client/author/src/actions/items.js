import {
  RECEIVE_ITEMS_REQUEST,
  RECEIVE_ITEM_REQUEST,
  CREATE_ITEM_REQUEST,
  UPDATE_ITEM_REQUEST
} from "../constants/actions";

export const receiveItemsAction = ({ page, limit, search }) => ({
  type: RECEIVE_ITEMS_REQUEST,
  payload: { page, limit, search }
});

export const receiveItemByIdAction = id => ({
  type: RECEIVE_ITEM_REQUEST,
  payload: { id }
});

export const createItemAction = payload => ({
  type: CREATE_ITEM_REQUEST,
  payload: { payload }
});

export const updateItemByIdAction = payload => ({
  type: UPDATE_ITEM_REQUEST,
  payload: { payload }
});
