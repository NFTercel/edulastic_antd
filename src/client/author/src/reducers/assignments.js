import {
  RECEIVE_ASSIGNMENTS_REQUEST,
  RECEIVE_ASSIGNMENTS_SUCCESS,
  RECEIVE_ASSIGNMENTS_ERROR
} from "../constants/actions";

const initialState = {
  entities: [],
  error: null,
  page: 1,
  limit: 20,
  count: 0,
  loading: false,
  creating: false
};

const reducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case RECEIVE_ASSIGNMENTS_REQUEST:
      return { ...state, loading: true };
    case RECEIVE_ASSIGNMENTS_SUCCESS:
      return {
        ...state,
        loading: false,
        entities: payload.entities
      };
    case RECEIVE_ASSIGNMENTS_ERROR:
      return { ...state, loading: false, error: payload.error };
    default:
      return state;
  }
};

export default reducer;
