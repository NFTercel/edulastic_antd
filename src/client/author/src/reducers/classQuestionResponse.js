import {
  RECEIVE_CLASS_QUESTION_REQUEST,
  RECEIVE_CLASS_QUESTION_SUCCESS,
  RECEIVE_CLASS_QUESTION_ERROR
} from "../constants/actions";

const initialState = {
  data: [],
  error: null,
  loading: false
};

const reducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case RECEIVE_CLASS_QUESTION_REQUEST:
      return { ...state, loading: true };
    case RECEIVE_CLASS_QUESTION_SUCCESS:
      return {
        ...state,
        loading: false,
        data: payload
      };
    case RECEIVE_CLASS_QUESTION_ERROR:
      return { ...state, loading: false, error: payload.error };
    default:
      return state;
  }
};

export default reducer;
