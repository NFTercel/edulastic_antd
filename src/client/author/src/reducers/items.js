import { combineReducers } from "redux";

import {
  RECEIVE_ITEMS_REQUEST,
  RECEIVE_ITEMS_SUCCESS,
  RECEIVE_ITEMS_ERROR,
  RECEIVE_ITEM_REQUEST,
  RECEIVE_ITEM_SUCCESS,
  RECEIVE_ITEM_ERROR
} from "../constants/actions";

const initialItemsState = {
  items: [],
  error: null,
  loading: false,
  page: 1,
  limit: 5,
  count: 0
};

export const itemsReducer = (state = initialItemsState, { type, payload }) => {
  switch (type) {
    case RECEIVE_ITEMS_REQUEST:
      return { ...state, loading: true };
    case RECEIVE_ITEMS_SUCCESS:
      return {
        ...state,
        loading: false,
        items: payload.items,
        page: payload.page,
        limit: payload.limit,
        count: payload.count
      };
    case RECEIVE_ITEMS_ERROR:
      return { ...state, loading: false, error: payload.error };
    default:
      return state;
  }
};

const initialItemState = {
  item: null,
  error: null,
  loading: false
};

export const itemReducer = (state = initialItemState, { type, payload }) => {
  switch (type) {
    case RECEIVE_ITEM_REQUEST:
      return { ...state, loading: true };
    case RECEIVE_ITEM_SUCCESS:
      return { ...state, loading: false, item: payload.item };
    case RECEIVE_ITEM_ERROR:
      return { ...state, loading: false, error: payload.error };
    default:
      return state;
  }
};

export default combineReducers({
  list: itemsReducer,
  item: itemReducer
});
