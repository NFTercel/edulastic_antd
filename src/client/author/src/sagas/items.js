import { takeEvery, takeLatest, call, put, all } from "redux-saga/effects";
import { itemsApi } from "@edulastic/api";
import { message } from "antd";

import {
  RECEIVE_ITEM_REQUEST,
  RECEIVE_ITEM_SUCCESS,
  RECEIVE_ITEM_ERROR,
  RECEIVE_ITEMS_REQUEST,
  RECEIVE_ITEMS_SUCCESS,
  RECEIVE_ITEMS_ERROR,
  CREATE_ITEM_REQUEST,
  UPDATE_ITEM_REQUEST
} from "../constants/actions";

export function* receiveItemsSaga({ payload }) {
  try {
    const { items, page, count } = yield call(itemsApi.receiveItems, payload);

    yield put({
      type: RECEIVE_ITEMS_SUCCESS,
      payload: { items, page, limit: payload.limit, count }
    });
  } catch (err) {
    console.error(err);
    const errorMessage = "Receive items is failing";
    yield call(message.error, errorMessage);
    yield put({
      type: RECEIVE_ITEMS_ERROR,
      payload: { error: errorMessage }
    });
  }
}

export function* receiveItemSaga({ payload }) {
  try {
    const item = yield call(itemsApi.receiveItemById, payload.id);

    yield put({
      type: RECEIVE_ITEM_SUCCESS,
      payload: { item }
    });
  } catch (err) {
    console.error(err);
    const errorMessage = "Receive item by id is failing";
    yield call(message.error, errorMessage);
    yield put({
      type: RECEIVE_ITEM_ERROR,
      payload: { error: errorMessage }
    });
  }
}

export function* createItemSaga({ payload }) {
  try {
    const item = yield call(itemsApi.createItem, payload);
    yield put({
      type: RECEIVE_ITEM_SUCCESS,
      payload: { item: item.data }
    });
  } catch (err) {
    console.error(err);
    const errorMessage = "Create item is failed";
    yield call(message.error, errorMessage);
    yield put({
      type: RECEIVE_ITEM_ERROR,
      payload: { error: errorMessage }
    });
  }
}

export function* updateItemSaga({ payload }) {
  try {
    const item = yield call(itemsApi.updateItemById, payload);
    yield put({
      type: RECEIVE_ITEM_SUCCESS,
      payload: { item: item.data }
    });
  } catch (err) {
    console.error(err);
    const errorMessage = "Update item is failed";
    yield call(message.error, errorMessage);
    yield put({
      type: RECEIVE_ITEM_ERROR,
      payload: { error: errorMessage }
    });
  }
}

export default function* watcherSaga() {
  yield all([
    yield takeEvery(RECEIVE_ITEM_REQUEST, receiveItemSaga),
    yield takeLatest(RECEIVE_ITEMS_REQUEST, receiveItemsSaga),
    yield takeLatest(CREATE_ITEM_REQUEST, createItemSaga),
    yield takeLatest(UPDATE_ITEM_REQUEST, updateItemSaga)
  ]);
}
