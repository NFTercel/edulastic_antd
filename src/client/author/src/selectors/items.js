import { createSelector } from "reselect";

export const stateSelector = state => state.items;
export const itemsSelector = createSelector(
  stateSelector,
  state => state.list
);
export const itemSelector = createSelector(
  stateSelector,
  state => state.item
);

export const getItemSelector = createSelector(
  itemSelector,
  state => state.item
);
export const getItemLoadingSelector = createSelector(
  itemSelector,
  state => state.loading
);

export const getItemsListSelector = createSelector(
  itemsSelector,
  state => state.items
);
export const getItemsPageSelector = createSelector(
  itemsSelector,
  state => state.page
);
export const getItemsLimitSelector = createSelector(
  itemsSelector,
  state => state.limit
);
export const getItemsCountSelector = createSelector(
  itemsSelector,
  state => state.count
);
export const getItemsLoadingSelector = createSelector(
  itemsSelector,
  state => state.loading
);
