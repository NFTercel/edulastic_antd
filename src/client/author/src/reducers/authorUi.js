import { TOGGLE_MENU, RESPONSIVE_TOGGLE_MENU } from "../constants/actions";

const initialState = {
  isSidebarCollapsed: window.innerWidth < 770 ? true : false,
  isResponsive: false
};
const reducer = (state = initialState, { type }) => {
  switch (type) {
    case TOGGLE_MENU:
      return {
        ...state,
        isSidebarCollapsed: !state.isSidebarCollapsed
      };
    case RESPONSIVE_TOGGLE_MENU:
      return {
        ...state,
        isResponsive: !state.isResponsive
      };
    default:
      return state;
  }
};

export default reducer;
