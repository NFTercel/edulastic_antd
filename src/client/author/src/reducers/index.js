import authorUi from "./authorUi";
import view from "./view";
import items from "./items";
import testItem from "./testItem";
import dictionaries from "./dictionaries";
import author_assignments from "./assignments";
import author_classboard_gradebook from "./gradeBook";
import author_classboard_testActivity from "./testActivity";
import authorGroups from "../../sharedDucks/groups";
import authorQuestions from "../../sharedDucks/questions";
import classResponse from "./classResponse";
import studentResponse from "./studentResponse";
import feedbackResponse from "./feedbackResponse";
import studentQuestionResponse from "./studentQuestionResponse";
import classQuestionResponse from "./classQuestionResponse";
import { itemAdd } from "../../ItemAdd";
import { testList } from "../../TestList";
import { tests } from "../../TestPage";
import { itemDetail } from "../../ItemDetail";
import { question } from "../../QuestionEditor";
import { testsAddItems } from "../../TestPage/components/AddItems";
import { testAssignmentsReducer } from "../../TestPage/components/Assign";
import { assessmentCreate } from "../../AssessmentCreate";
import { manageClass } from "../../ManageClass";

const authorReducers = {
  authorUi,
  view,
  authorGroups,
  items,
  testsAddItems,
  itemAdd,
  question,
  testItem,
  itemDetail,
  dictionaries,
  authorQuestions,
  author_assignments,
  authorTestAssignments: testAssignmentsReducer,
  author_classboard_gradebook,
  author_classboard_testActivity,
  classResponse,
  studentResponse,
  feedbackResponse,
  tests,
  testList,
  assessmentCreate,
  studentQuestionResponse,
  classQuestionResponse,
  manageClass
};

export default authorReducers;
