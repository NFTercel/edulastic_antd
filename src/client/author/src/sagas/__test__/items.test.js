import { expectSaga } from "redux-saga-test-plan";
import items from "../items";
import { itemsReducer, itemReducer } from "../../reducers/items";

import {
  RECEIVE_ITEMS_REQUEST,
  RECEIVE_ITEMS_SUCCESS,
  RECEIVE_ITEMS_ERROR,
  RECEIVE_ITEM_REQUEST,
  RECEIVE_ITEM_SUCCESS,
  RECEIVE_ITEM_ERROR
} from "../../constants/actions";

describe("items", () => {
  it("should have the expected watchers", done =>
    expectSaga(items)
      .run({ silenceTimeout: true })
      .then(saga => {
        expect(saga).toMatchSnapshot();
        done();
      }));

  it(`should handle ${RECEIVE_ITEMS_REQUEST}`, () => {
    expect(
      itemsReducer(undefined, {
        type: RECEIVE_ITEMS_REQUEST,
        payload: { page: 1, limit: 10, count: 0 }
      })
    ).toMatchSnapshot();
  });

  it(`should handle ${RECEIVE_ITEMS_SUCCESS}`, () => {
    expect(
      itemsReducer(undefined, {
        type: RECEIVE_ITEMS_SUCCESS,
        payload: { page: 1, limit: 10, count: 0 }
      })
    ).toMatchSnapshot();
  });

  it(`should handle ${RECEIVE_ITEMS_ERROR}`, () => {
    expect(
      itemsReducer(undefined, {
        type: RECEIVE_ITEMS_ERROR,
        payload: { error: "something went wrong" }
      })
    ).toMatchSnapshot();
  });

  it(`should handle ${RECEIVE_ITEM_REQUEST}`, () => {
    expect(
      itemReducer(undefined, {
        type: RECEIVE_ITEM_REQUEST,
        payload: { id: 1 }
      })
    ).toMatchSnapshot();
  });

  it(`should handle ${RECEIVE_ITEM_SUCCESS}`, () => {
    expect(
      itemReducer(undefined, {
        type: RECEIVE_ITEM_SUCCESS,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it(`should handle ${RECEIVE_ITEM_ERROR}`, () => {
    expect(
      itemReducer(undefined, {
        type: RECEIVE_ITEM_ERROR,
        payload: { error: "something went wrong" }
      })
    ).toMatchSnapshot();
  });
});
