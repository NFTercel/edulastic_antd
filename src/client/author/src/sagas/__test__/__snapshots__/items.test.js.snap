// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`items should handle [items] receive item error 1`] = `
Object {
  "error": "something went wrong",
  "item": null,
  "loading": false,
}
`;

exports[`items should handle [items] receive item request 1`] = `
Object {
  "error": null,
  "item": null,
  "loading": true,
}
`;

exports[`items should handle [items] receive item success 1`] = `
Object {
  "error": null,
  "item": undefined,
  "loading": false,
}
`;

exports[`items should handle [items] receive items error 1`] = `
Object {
  "count": 0,
  "error": "something went wrong",
  "items": Array [],
  "limit": 5,
  "loading": false,
  "page": 1,
}
`;

exports[`items should handle [items] receive items request 1`] = `
Object {
  "count": 0,
  "error": null,
  "items": Array [],
  "limit": 5,
  "loading": true,
  "page": 1,
}
`;

exports[`items should handle [items] receive items success 1`] = `
Object {
  "count": 0,
  "error": null,
  "items": undefined,
  "limit": 10,
  "loading": false,
  "page": 1,
}
`;

exports[`items should have the expected watchers 1`] = `
Object {
  "fork": Array [
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [
          "[items] receive item request",
          "@@redux-saga-test-plan/json/function/receiveItemSaga",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/takeEvery",
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [
          "[items] receive items request",
          "@@redux-saga-test-plan/json/function/receiveItemsSaga",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/takeLatest",
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [
          "[items] create item request",
          "@@redux-saga-test-plan/json/function/createItemSaga",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/takeLatest",
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "FORK": Object {
        "args": Array [
          "[items] update item request",
          "@@redux-saga-test-plan/json/function/updateItemSaga",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/takeLatest",
      },
    },
  ],
  "take": Array [
    Object {
      "@@redux-saga/IO": true,
      "TAKE": Object {
        "pattern": "[items] receive item request",
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "TAKE": Object {
        "pattern": "[items] receive items request",
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "TAKE": Object {
        "pattern": "[items] create item request",
      },
    },
    Object {
      "@@redux-saga/IO": true,
      "TAKE": Object {
        "pattern": "[items] update item request",
      },
    },
  ],
}
`;
